<launch>
    <arg name="episode_seq" value="1"/>
    <param name="/simulation_seq" value="$(arg episode_seq)" type="int"/>
    <!-- ATTENTION! SYNC these in algorithm.launch visualization.launch map_generator.launch -->
    <arg name="radius_comm" value="34.0" />
    <arg name="map_size_x" value="50.0"/>
    <arg name="map_size_y" value="50.0"/>
    <arg name="map_size_z" value="5.0"/>
    <arg name="grid_resolution" value="5.0"/>
    <!-- ATTENTION! SYNC these in algorithm.launch visualization.launch map_generator.launch -->

    <include file="$(find map_generator)/launch/map_generator.launch">
        <arg name="use_real_map" value="false"/>
        <arg name="read_map" value="false"/>

        <arg name="map_size_x_" value="$(arg map_size_x)"/>
        <arg name="map_size_y_" value="$(arg map_size_y)"/>
        <arg name="map_size_z_" value="$(arg map_size_z)"/>
        <arg name="grid_resolution" value="$(arg grid_resolution)"/>
        <arg name="c_num" value="0"/>
        <arg name="p_num" value="10"/>
    </include>

    <node pkg="visualization" name="entities_visualization" type="entities_visualization" output="screen">
        <param name="radius_comm" value="$(arg radius_comm)" type="double"/>

        <remap from="~target_pos" to="target_pos"/>
        <remap from="~uav_pos" to="uav_pos"/>
        <remap from="~bs_pos" to="bs_pos"/>
        <remap from="~solution" to="solution"/>

        <param name="uav_size" value="1.2"/>
        <param name="bs_size" value="24.0"/>
        <param name="target_size" value="0.6"/>

        <param name="edge_size" value="0.1"/>
        <param name="edge_a" value="0.8"/>
        <param name="weak_edge_a" value="0.0"/>

        <!--  Grid map parameters for LoS checker  -->
        <param name="los_checker/resolution" value="0.25" type="double"/>
        <param name="los_checker/x_size" value="$(arg map_size_x)" type="double"/>
        <param name="los_checker/y_size" value="$(arg map_size_y)" type="double"/>
        <param name="los_checker/z_size" value="$(arg map_size_z)" type="double"/>
        <remap from="/global_map" to="/map_generator/global_cloud"/>
        <!--  Grid map parameters for LoS checker  -->
    </node>

    <node pkg="visualization" name="scp_visualization" type="scp_visualization" output="screen">
        <param name="scp_a" value="0.6"/>
        
    </node>

    <node name="rviz_visualisation" pkg="rviz" type="rviz" output="log"
          args="-d $(find visualization)/config/visualization.rviz">
    </node>

</launch>
